core/src/main/kotlin/org/evomaster/core/problem/rest/classifier/ModelMetrics.kt

```diff
@@ -0,0 +1,99 @@
+package org.evomaster.core.problem.rest.classifier
+
+/**
+ * Interface for tracking and estimating performance metrics of a model
+ * that predicts HTTP response status codes for REST calls.
+ *
+ * Implementations are expected to maintain internal counters for the confusion matrix:
+ * - TP (True Positives): predicted 400 and actual 400
+ * - TN (True Negatives): predicted not-400 and actual not-400
+ * - FP (False Positives): predicted 400 but actual not-400
+ * - FN (False Negatives): predicted not-400 but actual 400
+ *
+ * From these values, various metrics can be derived, such as:
+ * - Accuracy
+ * - Precision (for 400s)
+ * - Recall (for 400s)
+ * - F1 score (for 400s)
+ * - MCC (Matthews Correlation Coefficient, for 400 vs. not-400 classification)
+ */
+interface ModelMetrics {
+
+    /**
+     * Return a bundle of all key performance metrics as a [ModelEvaluation].
+     *
+     * This is the preferred way to query the model’s performance,
+     * as it provides a single unified object instead of calling
+     * individual metric methods.
+     */
+    fun estimateMetrics(): ModelEvaluation
+
+    /**
+     * Estimate the overall accuracy of the model.
+     *
+     * Accuracy = (TP + TN) / (TP + TN + FP + FN)
+     *
+     * The proportion of correct predictions
+     * (both positives and negatives) out of all evaluated predictions.
+     */
+    fun estimateAccuracy(): Double
```

Comment:
isn't this just `return estimateMetrics().accuracy`? if so, could provide it here as `default` implementation for this interface. or am i missing something? also, this comment applies to all other functions here in this interface

Reply:
or, alternatively, it should be `fun estimateMetrics()` that should have a default implementation calling the other methods (this might be a better option)